tests/testthat/test_msnset_se_coercions.R

```diff
@@ -21,6 +21,6 @@ test_that("se2msnset conversion", {
 	expect_equal(colnames(msnset_back), colnames(se))
 	expect_equal(rownames(msnset_back), rownames(se))
 	expect_equal(Biobase::exprs(msnset_back), SummarizedExperiment::assay(se))
-	expect_equal(Biobase::fData(msnset_back), data.frame(SummarizedExperiment::rowData(se), row.names = se@NAMES))
+	expect_equal(Biobase::fData(msnset_back), data.frame(SummarizedExperiment::rowData(se), row.names = names(se)))
```

Comment:
I missed this one before. Why do you compare against `msnset_back` and not directly against `msnset`? And wouldn't be `expect_equal(msnset, as(se, "MSnSet"))` enough?

And the `::` shouldn't be needed here.

Reply:
One issue with the current coercion is that it only converts the assayData, phenoData and featureData. The other slots, such as qual, processingData, experimentData, protocolData and annotation are _MSnSet_ specific. The _SummarizedExperiment_ supports a simple list as metadata, so one solution could be to parse all _MSnSet_ specific slots into a list which is stored in the _SummarizedExperiment_ metadata and vice versa for _SE_ to _MSnSet_ conversion. The question is whether it is necessary to transfer this additional information and whether it is preferred to store this information in a simple list?

Reply:
I will push the necessary `as(x, "list")` to `MSnbase` in the coming days, so that you can add these slots to the `SummarizedExperiment`'s `metadata` slot. Will update this comment with more instructions when the code becomes available.

Reply:
I have added the three `as` methods in the latest `MSnbase` on github. There is also `MSnbase:::.reduce_list` that will remove empty elements or those with and empty character (`""`) - `MSnbase:::.reduce_list(as(processingData(x), "list"))`, so as to avoid adding missing data. The names of the elements could possibly be prefixed with `MSnbase` to make it explicit where these metadata originate from.

Reply:
@lgatto: Thanks. In the end, I didn't use these functions as I could store the _data.frame_, _MSnProcess_, _MIAPE_ and _AnnotatedDataFrame_ objects in the metadata _list_ of SummarizedExperiment.

The current implementation is a bit 'MSnSet-focused', because it will correctly coerce all MSnSet slots into SE and back to MSnSet. However, SE metadata that does not fit into one of the MSnSet slots will be lost in the SE to MSnSet coercion.

Reply:
The issue with this is that once you have an object created with 
```r
se <- as(msnset, "SummarizedExperiment")
```
you will still need `MSnbase` to use `se` because of the `MSnProcess` and `MIAPE` objects in the metadata (`AnnotatedDataFrame` is defined in `Biobase`, which is a dependency of `SummarizedExperiment`). Having that assumption simplifies the code, but it limits the portability of that `se` object. @arnesmits - what is your typical use case?

Reply:
I will work on this next week, and only transfer some information as a list to the `SummarizedExperiment` metadata slot, and document the behaviour in the manual page.